tests/default.nix

```diff
@@ -0,0 +1,114 @@
+{
+  inputs,
+  lib,
+  self,
+  ...
+}:
+{
+  perSystem =
+    { pkgs, system, ... }:
+    let
+      nixos-lib = import (pkgs.path + "/nixos/lib") { };
+    in
+    {
+      checks =
+        lib.flip lib.mapAttrs (lib.filterAttrs (_: type: type == "directory") (builtins.readDir ./cases))
+          (
+            name: _:
+            let
+              testFlake = ./cases/${name};
+            in
+            (nixos-lib.runTest {
+              hostPkgs = pkgs;
+              name = "agenix-rekey-test-${name}";
+              #test = ../examples/flake-parts;
+              # This speeds up the evaluation by skipping evaluating documentation (optional)
+              defaults.documentation.enable = lib.mkDefault false;
+              # This makes `self` available in the NixOS configuration of our virtual machines.
+              # This is useful for referencing modules or packages from your own flake
+              # as well as importing from other flakes.
+              node.specialArgs = { inherit self; };
+              nodes.node =
+                { pkgs, ... }:
+                {
+                  environment.etc.setupScript = {
+                    mode = "777";
+                    source = pkgs.writeText "test" ''
+                      cp -r ${testFlake} /tmp/test
+                      chmod -R 777 /tmp/test
+                      cd /tmp/test
+                      cp ${./ssh_priv.key} /etc/ssh/ssh_host_rsa_key
+                      chmod 666 /etc/ssh/ssh_host_rsa_key
+                      export NIX_CONFIG="extra-experimental-features = flakes nix-command"
+                      nix flake lock --offline --override-input agenix-rekey ${../.} ${
```

Comment:
```suggestion
                      # Make sure to override all inputs to the exact input we have in our main testing flake to prevent any downloads in the VM
                      nix flake lock --offline --override-input agenix-rekey ${../.} ${
```